packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1058,3 +1065,273 @@ class CallbackShortcuts extends StatelessWidget {
     );
   }
 }
+
+/// A token returned by [ShortcutRegistry.addAll] that allows the caller to
+/// identify the shortcuts they registered with the [ShortcutRegistry] through
+/// the [ShortcutRegistrar].
+///
+/// When the token is no longer needed, [dispose] should be called, and the
+/// token should no longer be used.
+class ShortcutRegistryToken {
+  // Tokens can only be created by the ShortcutRegistry.
+  const ShortcutRegistryToken._(this.registry);
+
+  /// The [ShortcutRegistry] that this token was issued by.
+  final ShortcutRegistry registry;
+
+  /// Called when the token is no longer needed.
+  ///
+  /// Call this will remove all shortcuts associated with this
+  /// [ShortcutRegistryToken] from the [registry].
+  @mustCallSuper
+  void dispose() {
+    registry._disposeToken(this);
+  }
+}
+
+/// A specialized [ShortcutManager] used by [ShortcutRegistrar] that allows
+/// adding or removing shortcut bindings by descendants of the
+/// [ShortcutRegistrar].
+///
+/// You can reach the nearest [ShortcutRegistry] using [of] and [maybeOf].
+///
+/// The registry may be listened to (with [addListener]/[removeListener]) for
+/// change notifications when the registered shortcuts change.
+class ShortcutRegistry extends ShortcutManager {
+  /// Gets the combined shortcut bindings from all contexts that are registered
+  /// with this [ShortcutRegistry], in addition to the bindings passed to
+  /// [ShortcutRegistry.ShortcutRegistry].
+  ///
+  /// Listeners will be notified when the value returned by this getter changes.
+  ///
+  /// Returns a copy: modifying the returned map will have no effect.
+  @override
+  Map<ShortcutActivator, Intent> get shortcuts {
+    assert(debugAssertNotDisposed());
+    return <ShortcutActivator, Intent>{
+      ..._shortcuts,
+      for (final MapEntry<ShortcutRegistryToken, Map<ShortcutActivator, Intent>> entry in _tokenShortcuts.entries)
+        ...entry.value,
+    };
+  }
+  final Map<ShortcutRegistryToken, Map<ShortcutActivator, Intent>> _tokenShortcuts =
+    <ShortcutRegistryToken, Map<ShortcutActivator, Intent>>{};
+
+  /// Adds all the given shortcut bindings to this [ShortcutRegistry], and
+  /// returns a token for managing those bindings.
+  ///
+  /// The token should have [ShortcutRegistryToken.dispose] called on it when
+  /// these shortcuts are no longer needed. This will remove them from the
+  /// registry, and invalidate the token.
+  ///
+  /// This method will assert in debug mode if another token exists (i.e. hasn't been
+  /// disposed of) has already added a given shortcut.
```

Comment:
I think this sentence is grammatically not correct? Maybe missing a "that" before "has already"?